scripts/generate-rule-ids/main.py

```diff
@@ -18,35 +18,43 @@ def generate_rule_ids():
     """
     Generates rule IDs for all rules that don't already have one. This is done by generating a UUID from the rule's
     file path. This is deterministic, so the same file path will always generate the same UUID.
+
+    Also validates that not rule IDs are duplicated.
     """
+    known_ids = set()
+
     for directory in RULE_DIRS:
         for root, dirs, files in os.walk(directory):
             for file in files:
+                # CI validates that no .YAML files exist
                 if file.endswith('.yml'):
                     # Read the file contents
                     full_file_path = os.path.join(root, file)
                     with open(full_file_path, 'r') as f:
                         contents = f.read()
 
-                    try:
                         # load the YAML and check if it already has an ID
                         parsed = yaml.safe_load(contents)
                         feed_rule_id = parsed.get('id')
 
                         if feed_rule_id:
+                            # It's expected for this ID not to always match the path generated ID. The purpose of the
```

Comment:
It'd be sweet if we could check if the PR itself was trying to edit the ID too

Reply:
I might try and add something like that later, but it's a bit more complex and might have some edge cases with how Git recognizes changes.